app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Support\Facades\Storage;
+use App\Http\Requests\UpdateUserRequest;
+use App\Repositories\User\UserRepositoryInterface;
+
+class UserController extends Controller
+{
+    protected $userRepository;
+
+    /**
+     * UserController construct
+     * 
+     * UserRepositoryInterface $userRepository
+     * @return void
+     */
+    public function __construct(
+        UserRepositoryInterface $userRepository
+    ) {
+        $this->userRepository = $userRepository;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return define response
+     */
+    public function index()
+    {
+        $users = $this->userRepository->getAll();
+
+        return response()->json($users);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * int  $id
+     * @return define response
+     */
+    public function show($id)
+    {
+        try {
+            $user = $this->userRepository->find($id);
```

Comment:
Use `first` or add limit 1 so that when the id is not PK or unique, the API returns only one record. Do the same with edit/update/delete.

Reply:
@phuongnn-0944  Trường id đang là khóa chính thì có cần thiết phải dùng first không anh?